packages/cli/generation/ir-generator/src/generateIntermediateRepresentation.ts

```diff
@@ -727,3 +731,49 @@ function filterServiceTypeReferenceInfoForAudiences(
         typesReferencedOnlyByService: filteredTypesReferencedOnlyByService
     };
 }
+
+type NamedObjectProperty = ObjectProperty & { valueType: TypeReference.Named };
+function markInlineTypes(types: Record<string, TypeDeclaration>) {
+    // find types that have properties containing inline types
+    const namedProps = Object.values(types).flatMap((type) => {
+        if (type.shape.type !== "object") {
+            return [];
+        }
```

Comment:
what happens if an inline type is "extended" or if a discriminated union references the inline type as a variant?

Reply:
I think you need to iterate over every single TypeReference and store the reference. Maybe we make `parseTypeReference` stateful? 

https://github.com/fern-api/fern/blob/cf894f08ab4d3f0dad59f4512a3c822ac558530e/packages/cli/generation/ir-generator/src/converters/type-declarations/convertObjectTypeDeclaration.ts#L36C24-L36C67